### ISSUE TITLE
Issue with scenes

### ISSUE THREAD

**Issue by soywiz** [korlibs/korge#144]
It seems that cleanup might not be working correctly somehow.

Sample from slack:

```kotlin
suspend fun main() = Korge(Korge.Config(module = MyModule))
object MyModule : Module() {
   override val mainScene: KClass<out Scene> = MyScene1::class
   override suspend fun init(injector: AsyncInjector): Unit = injector.run {
      mapInstance(MyDependency("HELLO WORLD"))
      mapPrototype { MyScene1(get()) }
      mapPrototype { MyScene2(get()) }
   }
}
class MyDependency(val value: String)
class MyScene1(val myDependency: MyDependency) : Scene() {
   override suspend fun Container.sceneInit() {
      text("MyScene1: ${myDependency.value}") { filtering = false }
      solidRect(100, 100, Colors.RED) {
         position(200, 200)
         onClick {
            sceneContainer.changeTo<MyScene2>()
            println("CLICKED ON SC1")
         }
      }
   }
}
class MyScene2(val myDependency: MyDependency) : Scene() {
   override suspend fun Container.sceneInit() {
      text("MyScene2: ${myDependency.value}") { filtering = false }
      solidRect(100, 100, Colors.BLUE) {
         position(400, 200)
         onClick {
            sceneContainer.changeTo<MyScene1>(MyDependency("From MyScene2"))
         }
      }
   }
}
```

**Comment by tsaulic** [comment]
Just in case context of this bug gets forgotten, after changing to scene 2, the scene 1 button is still clickable in this example.

**Comment by soywiz** [comment]
I think this was fixed on 2.0 here: https://github.com/korlibs/korge-next/commit/924a95f139997615a324d338c2d045025388f4e6
